lib/matplotlib/backends/backend_wx.py

```diff
@@ -810,12 +790,41 @@ def _onPaint(self, evt):
         """
 
         DEBUG_MSG("_onPaint()", 1, self)
-        drawDC = wx.PaintDC(self)
         if not self._isDrawn:
-            self.draw(drawDC=drawDC)
+            self._draw()
+
+        # the bitmap can not be in use by another DC
+        img = self.bitmap.ConvertToImage()
+        bmp = img.ConvertToBitmap()
+        dc = wx.BufferedPaintDC(self, bmp)
+
+        if not self._rubberband:
+            return
+
+        # draw rubberband / selection using an overlay
+        if self._overlay is None:
+            self._overlay = wx.Overlay()
+        odc = wx.DCOverlay(self._overlay, dc)
+
+        if not self._retinaFix:
+            # draw a box with border and 50% transparency
+            dc = wx.GCDC(dc)
+            # Set the pen, for the box's border
+            bc = wx.BLUE
+            dc.SetPen(wx.Pen(colour=bc, width=1, style=wx.PENSTYLE_SOLID))
+            # Create a brush (for the box's interior) with the same colour,
+            # but 50% transparency.
+            bc = wx.Colour(bc.red, bc.green, bc.blue, 0x80)
+            dc.SetBrush(wx.Brush(bc))
         else:
-            self.gui_repaint(drawDC=drawDC)
-        drawDC.Destroy()
+            # draw without transparency which is buggy on Retina displays
+            dc.SetPen(wx.Pen(wx.BLACK, 1, wx.PENSTYLE_SHORT_DASH))
+            dc.SetBrush(wx.TRANSPARENT_BRUSH)
+
```

Comment:
some comments and suggestions:

a) use   dc.SetLogicalFunction(wx.XOR) and a white (or light) Pen color so that the border shows on axes with different background face colors.  Similarly, let the brush be more transparent.

b) use  the explicit `'wxMac' in wx.PlatformInfo` -- it's only used once anyway.

c) why is `dc` overwritten with `wx.GCDC(dc)`?  That seems unnecessary.


Perhaps something like this:

```
        odc = wx.DCOverlay(self._overlay, dc)

        dc.SetLogicalFunction(wx.XOR)
        dc.SetPen(wx.Pen(colour=wx.WHITE, width=2,
                         style=wx.PENSTYLE_SOLID))

        # Create a semi-transparent brush for the box's interior
        # except on Macs, as that is buggy on Retina displays
        if 'wxMac' in wx.PlatformInfo:
            dc.SetBrush(wx.TRANSPARENT_BRUSH)
        else:
            dc.SetBrush(wx.Brush(wx.Colour(255, 255, 0, 0x20)))

        # Draw the rectangle
```